R/Private.R

```diff
@@ -0,0 +1,88 @@
+.getPreferenceScoreDist <- function(connectionDetails = NULL,
+                                    preferenceScoreDist = NULL,
+                                    targetId,
+                                    comparatorId,
+                                    databaseId,
+                                    analysisId) {
+  if (!is.null(connectionDetails)) {
+    if (exists("connectionDetails@schema")) {
+      schema <- connectionDetails@schema
+    } else {
+      schema <- NULL
+    }
+    connection <- DatabaseConnector::connect(connectionDetails)
+    sql <-   "SELECT *
+              FROM  {@schema !=} ? {@schema.preference_score_dist}:{preference_score_dist}
+              WHERE target_id = @targetId
+            	AND comparator_id = @comparatorId
+            	AND database_id = '@databaseId'
+            	AND analysis_id = @analysisId;"
+    
+    sql <-
+      SqlRender::render(
+        sql = sql,
+        targetId = targetId,
+        comparatorId = comparatorId,
+        schema = schema,
+        databaseId = databaseId,
+        analysisId = analysisId
+      )
+    sql <-
+      SqlRender::translate(sql = sql, targetDialect = connection@dbms)
+    preferenceScoreDist <-
+      DatabaseConnector::querySql(
+        connection = connection,
+        sql = sql,
+        snakeCaseToCamelCase = TRUE
+      )
+  } else {
+    preferenceScoreDist <- preferenceScoreDist %>%
+      dplyr::filter(
+        targetId %in% !!targetId,
+        comparatorId %in% !!comparatorId,
+        databaseId %in% !!databaseId,
+        analysisId %in% !!analysisId
+      ) %>%
+      dplyr::select(preferenceScore, targetDensity, comparatorDensity)
+  }
+  return(preferenceScoreDist)
+}
+
+
+.getExposureOfInterest <- function(connectionDetails = NULL,
+                                   exposureOfInterest = NULL,
+                                   targetId ,
+                                   comparatorId) {
+  if (!is.null(connectionDetails)) {
+    if (exists("connectionDetails@schema")) {
```

Comment:
I'm sorry, I really should have deprecated the schema attribute of connectionDetails by now. Please don't use it. Instead, require the user to provide a databaseSchema argument in addition to connectionDetails. 

(Note that we call it 'databaseSchema', because the different database platforms use 'database' and 'schema' in different ways, and for SQL Server we combine 'database' and 'schema' into this single string)

Reply:
@schuemie  please confirm if i understood this right
- require user to provide databaseSchema as part of the function call getPreferenceScore.
- ignore the parameter schema in DatabaseConnector::createConnectionDetails

Sounds like we need to use databaseSchema in every (similar) planned function e.g. getPower. We probably need good and consistent documentation to explain this parameter for each of this function. I found this as documentation for this parameter for getTableNames function. I dont think it is very clear. Also, should we do checks/assertion for this parameter?

(instead of duplicating, maybe this should be a parameter in databaseConnector?)

```
The databaseSchema argument is interpreted differently according to the different platforms: SQLServer and PDW: The databaseSchema schema should specify both the database and the schema,e.g. ’my_database.dbo’. PostgreSQL and Redshift: The databaseSchema should specify the schema.Oracle:  The databaseSchema should specify the Oracle ’user’.  MySql and Impala:  The databas-eSchema should specify the database.
```

Reply:
Every R function we have that touches a database has one or more databaseSchema arguments. For example, [getDbCohortMethodData](https://ohdsi.github.io/CohortMethod/reference/getDbCohortMethodData.html) has three! (And that is me ignoring people who want vocabularyDatabaseSchema to be a separate argument). I don't want those in the connectionDetails because there can be multiple relevant databaseSchemas, and because DatabaseConnector is supposed to be application-agnostic. 

I agree the documentation isn't the best. But I also haven't heard complaints about this specific issue. Most people know what the databaseSchema is on their platform.

One tip: if you find you need to document the same argument many times, you can use roxygen templates. For example, I document a parameter once [here](https://github.com/OHDSI/CohortDiagnostics/blob/master/man-roxygen/CdmDatabaseSchema.R), and use that everywhere like [this](https://github.com/OHDSI/CohortDiagnostics/blob/master/R/CohortConstruction.R#L109).

Reply:
Thank you! I will add assertCheck for @databaseSchema